ctgan/synthesizers/ctgan.py

```diff
@@ -232,6 +233,11 @@ def fit(self, train_data, discrete_columns=tuple()):
                 contain the integer indices of the columns. Otherwise, if it is
                 a ``pandas.DataFrame``, this list should contain the column names.
         """
+        if epochs:
+            warnings.warn("Passing 'epochs' to 'fit' will be deprecated in future versions. "
+                          "Please pass it to the class constructor instead.", FutureWarning)
+            self._epochs = epochs
```

Comment:
Let's invert the schema: Instead of assigning `epochs` to `self._epochs` if `epochs` is given, let's assign `self._epochs` to `epochs` if it is not given (and later on use `epochs` instead of `self._epochs`).
This way, the instance attributes are not modified.